changes/metadata.go

```diff
@@ -0,0 +1,225 @@
+package changes
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+)
+
+// SchemaVersion defines the current JSON schema version for persistent data types (Change, Release, ...)
+const SchemaVersion = "1.0"
+
+// Metadata is a representation of the change metadata stored in a .changes directory.
+type Metadata struct {
+	ChangePath string     // ChangePath is the relative path from the current directory to .changes
+	Changes    []*Change  // Changes holds all pending change metadata in the .changes/next-release directory
+	Releases   []*Release // Releases contains all releases in the .changes/releases directory
```

Comment:
One thing to consider is the potential growth rate of the number of releases we do perform per year. Likely we don't want/need to load all the history about releases into memory. We should probably consider moving this to a  separate method that we can use to discover the release files, and lazily load their contents if we want to inspect them.